File: dato.config.js
const inspect = x => JSON.stringify(x, null, 2);
const htmlTag = require("html-tag");

const toHtml = tags =>
  tags
    .map(({ tagName, attributes, content }) =>
      htmlTag(tagName, attributes, content)
    )
    .join("");

const toSlug = str =>
  str
    .trim()
    .replace(/[^a-zA-Z0-9]+/g, "-")
    .toLowerCase();

// dato.config.js
module.exports = (dato, root, i18n) => {
  const { home, tabs } = dato;

  console.log("Collections:", Object.keys(dato.collectionsByType));

  // root.directory('content', content => {
  // content.createPost(`/_index.md`, 'yaml', {
  //   frontmatter: {
  //     title: '',
  //     seoMetaTags: toHtml(home.seoMetaTags)
  //   },
  //   content: home.content
  // });

  // ---------- lising type (midwife) ----------
  const tabMaps = tabs
    .map(({ title, image, blurb, slug, hideOnGentleBirth }) => ({
      title,
      image,
      blurb,
      slug,
      hideOnGentleBirth
    }))
    .filter(({ hideOnGentleBirth }) => hideOnGentleBirth === false);

  root.createDataFile("/_data/listings.json", "json", tabMaps);
  // make a new folder for each tab
  //   content.directory(tab.slug, dir => {
  //     // ---------- listing (Karla) ----------
  //   //   tab.listings.map((listing, i) => {
  //   //     dir.createPost(`/${toSlug(listing.name)}.md`, 'yaml', {
  //   //       frontmatter: {
  //   //         title: listing.name,
  //   //         photo:
  //   //           listing.picture &&
  //   //             listing.picture.url({
  //   //               auto: 'compress',
  //   //               w: 300,
  //   //               fm: 'jpg'
  //   //             }),
  //   //         type: 'tab',
  //   //         category: tab.title,
  //   //         categorySlug: tab.slug,
  //   //         email: listing.email,
  //   //         phone: listing.phone,
  //   //         websiteUrl: listing.website,
  //   //         seoMetaTags: toHtml(listing.seoMetaTags),
  //   //         weight: i
  //   //       },
  //   //       content: listing.description
  //   //     });
  //   //   });
  //   // });
  //
  //   console.log(JSON.stringify(tab.toMap(), null, 4));
  // });
  // });
};

File: dato.config.rb
# frozen_string_literal: true
create_data_file(
  '_data/settings.yml',
  :yaml,
  favicon_meta_tags: dato.site.favicon_meta_tags
)

# # inside a "_posts" directory...
# directory '_posts' do
#   # ...iterate over the "Blog post" records...
#   dato.posts.each do |article|
#     # ...and create a markdown file for each article!
#     create_post "#{article.date}-#{article.slug}.md" do
#       frontmatter(
#         :yaml,
#         title: article.title,
#         date: article.date || article.updated_at,
#         permalink: "/#{article.slug}",
#         layout: 'post',
#         seo_meta_tags: article.seo_meta_tags
#       )
#
#       content(article.content)
#     end
#   end
# end


homepage, *pages = dato.pages

create_post "index.md" do
  frontmatter(
    :yaml,
    title: homepage.title,
    permalink: '/',
    layout: 'page',
    seo_meta_tags: homepage.seo_meta_tags,
    body: homepage.content.to_hash,
    position: homepage.position
  )
end

# inside a "_posts" directory...
directory '_pages' do
  pages.each do |page|
    create_post "#{page.slug}.md" do
      frontmatter(
        :yaml,
        title: page.title,
        permalink: "/#{page.position == 1 ? "/" : page.slug}",
        layout: 'page',
        seo_meta_tags: page.seo_meta_tags,
        body: page.content.to_hash,
        position: page.position
      )
    end
  end
end
